### PR TITLE
feat: provide an option to deploy statically all webhook resources

### LINKED ISSUE #1014: Provide an options to deploy statically the webhooks
#### What would you like to be added:

In a more security restricted environment, there is a requirement to remove the permissions for `admissionregistration.k8s.io` from `security-profile-operator` cluster role in order to reduce the attack surface. This is not achievable with the current approach which dynamically creates/updates the webhook related resources. These API actions need these permissions. An alternative will be to create the webhook related resources statically as part of the operator deployment and skip the dynamic creation. It seems that they do not have dynamic dependencies.

This feature will introduce a running mode where the webhook resources are statically deploy outside of the operator. Most likely this will require a configuration option in the SPOD, in which case the operator will skip the creation and update of webhook related resources. 

#### Why is this needed:

This is needed to remove the `admissionregistration.k8s.io`  permissions from `security-profiles-operator` cluster role.

### PR DESCRIPTION
**Pull request by ccojocar**
#### What type of PR is this?

/kind feature 

#### What this PR does / why we need it:

It provides an option to deploy statically all webhook resources. It this mode the operator will not manage any webhook related resources but it will expect that all are already in place at deployment time. This allowed to reduce some cluster-wide permissions on RBAC role. 

This PR also slight refactored the e2e tests to allow the configuration of operator manifest.  This was required in order to be able to configured the static webhook manifest instead of standard operator. 
#### Does this PR have test?

Yes, but the e2e tests are not yet enabled in the CI. 

#### Does this PR introduce a user-facing change?

```release-note
Add an option to deploy statically all webhook resources in order to improve the operator security.
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
@ccojocar please rebase to get the latest CI changes in :)

**Comment by ccojocar** [comment]
@saschagrunert @janetkuo @JAORMX All tests passed. Any chance you can review this pull request? Thanks

**Comment by janetkuo** [comment]
I'm not familiar with this project so you probably want to find someone else to review it.

**Comment by ccojocar** [comment]
@janetkuo sorry for the noise. I picked up your handler by mistake.

File: hack/ci/e2e-static-webhook.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+# Copyright 2021 The Kubernetes Authors.
```
Comment (saschagrunert) [inline hack/ci/e2e-static-webhook.sh:2]: Nit:
```suggestion
# Copyright 2022 The Kubernetes Authors.
```

**Review by saschagrunert — APPROVED** [review]
Just a non-blocking nit.

**Comment by saschagrunert** [comment]
Well done!

**Pull request merged**